From b76b194b96a46c710f73ba4130002e092c2e4d38 Mon Sep 17 00:00:00 2001
From: Rex Lu <rex.lu@mediatek.com>
Date: Fri, 8 Nov 2024 15:05:23 +0800
Subject: [PATCH 251/277] fixup! mtk: mt76: mt7996: mt7996 support extra
 option_type

adjust Outstanding Number to 0x2 for fix issue UL tput issue
when 2 port pcie are in different speed(e.g. P0=2L, P1=1L)
and RXD/RXP are sent separately in different pcie port.
Without this patch, the uplink t-put would have serious degradation.
Typical SKU has this issue is 7988D+mt7996 in pcie option type=2

Signed-off-by: Rex Lu <rex.lu@mediatek.com>
---
 mt7996/dma.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mt7996/dma.c b/mt7996/dma.c
index db29704a..61e5da75 100644
--- a/mt7996/dma.c
+++ b/mt7996/dma.c
@@ -462,10 +462,10 @@ static void mt7996_dma_enable(struct mt7996_dev *dev, bool reset)
 			   (dev->hif2->speed == PCIE_SPEED_8_0GT && dev->hif2->width < 2)) {
 			mt76_rmw(dev, WF_WFDMA0_GLO_CFG_EXT0 + hif1_ofs,
 				 WF_WFDMA0_GLO_CFG_EXT0_OUTSTAND_MASK,
-				 FIELD_PREP(WF_WFDMA0_GLO_CFG_EXT0_OUTSTAND_MASK, 0x3));
+				 FIELD_PREP(WF_WFDMA0_GLO_CFG_EXT0_OUTSTAND_MASK, 0x2));
 			mt76_rmw(dev, MT_WFDMA_AXI_R2A_CTRL2,
 				 MT_WFDMA_AXI_R2A_CTRL2_OUTSTAND_MASK,
-				 FIELD_PREP(MT_WFDMA_AXI_R2A_CTRL2_OUTSTAND_MASK, 0x3));
+				 FIELD_PREP(MT_WFDMA_AXI_R2A_CTRL2_OUTSTAND_MASK, 0x2));
 		}
 		/* WFDMA rx threshold */
 		mt76_wr(dev, MT_WFDMA0_PAUSE_RX_Q_45_TH + hif1_ofs, 0xc000c);
-- 
2.45.2

